### PR TITLE
Feature/awesome accessibility color audit

### PR DESCRIPTION
**Pull request by khaykov**
Accessibility color audit turned into Calypsofication of colors.

I was surprised to learn that our styles and colors are total mess, specifically preference screens. In our app we have preferences that look like this :
[https://gyazo.com/2a6ce485fa9a8666ed7c66bf1793114f](https://gyazo.com/2a6ce485fa9a8666ed7c66bf1793114f)

With dialogs that looks like this:
[https://gyazo.com/42154b449f71c4cf058b2b1ae6eab3b3](https://gyazo.com/42154b449f71c4cf058b2b1ae6eab3b3)

So apart from adressing color contrast issues I tried to adress inconsistencies like that.

Now, there are parts where I wasn't exatly sure what colors to use. Reader in calypso, for example, uses low contrast icons, contrary to the rest of the interface, so I had to improvise.

So, I need a help of a designer with a sharp eye, like @iamthomasbishop, to help me navigate this _50 shades of grey_ ;)

I compiled before-after screenshots of major changes, so please let me know if you feel like we can use any other colors at any element.
[Part 1](https://user-images.githubusercontent.com/728822/38817390-3d9bad3e-41d3-11e8-8a66-3d944e04a0b3.jpg)
[Part 2](https://user-images.githubusercontent.com/728822/38817394-3dc76d84-41d3-11e8-8f8a-776c9b676ca1.jpg)
[Part 3](https://user-images.githubusercontent.com/728822/38817395-3df97680-41d3-11e8-940f-1e11596d34a7.jpg)
[Part 4](https://user-images.githubusercontent.com/728822/38817397-3e2abf60-41d3-11e8-9a48-514a1fd500b1.png)
[Part 5](https://user-images.githubusercontent.com/728822/38817399-3e5415f4-41d3-11e8-9970-c2589eacb8a9.png)

I haven't touched the login icons color, as there is no direct equivalent in Calypso. There are no contrast issues there, apart from icons. Should I change thir color to something greyer?
screenshot_1523892051

PS: Accessibility monitor is still angry about contrast of some colors, but I did my best to bring it at least above 3 (?).

PPS: Ringtone prefference is using system colors.

PPPS: I probably missed some colors.

PPPPS: We need to get our style game together.

### CODE REVIEW

**Comment by khaykov** [comment]
@folletto thanks for the insight! What I also mean by getting our style game together is standartizing and developing a proper style sheet, so we won't have to manually set a color or a text size to elements anymore. Looking at both android app and calypso I see that there is an affort to achieve this, but it's quite localized :)

@iamthomasbishop Thanks for a great review! After looking at all these greys I stopped noticing things :D I fixed most of the problems, and also have a couple of comments:

> If we decide to darken the text on the timestamp, we should also darken the icon to match it

Done!
[https://gyazo.com/cddb518ae100c0df1f7ded8c78585764](https://gyazo.com/cddb518ae100c0df1f7ded8c78585764)

Text placeholder color:
[https://gyazo.com/aab93a39f73c8c5d32c866c8df257fdc](https://gyazo.com/aab93a39f73c8c5d32c866c8df257fdc)

[https://gyazo.com/19c4c1d496ca1dfa2468a68688eae6d0](https://gyazo.com/19c4c1d496ca1dfa2468a68688eae6d0)

[https://gyazo.com/e4456e9a4c566e2fa97905d899919851](https://gyazo.com/e4456e9a4c566e2fa97905d899919851)

I found that `grey_darken_10` (#668eaa) works pretty well.  WDYT?

> Also, we could probably use that same (light) color on the inactive SEND button, and then when it's actionable, make it darker or some shade of blue. 

SEND button is always active, on Android at least.

> I like the darkening of the hint text below the inputs, and darkening the dropdown label 👍 . Perhaps the dropdown arrow icon should be the same color as the text - or - a darker shade of blue. I'd lean towards the former. One tiny thing: for the 500 characters remaining hint inside the textarea, can we either keep it the same as it was or darken it slightly? It seems to be lighter than previous

[https://gyazo.com/65fc750deb6e44a16e4d54cffb794906](https://gyazo.com/65fc750deb6e44a16e4d54cffb794906)

We are using `blue_wordpress` (#0087be) for similair controls, so I changed dropdown icon color to it.
`500 characters remaining` is now same as the hint color in other text input fields.

> Is there a brand shade in between the lighter color of the 'before' and darker of 'after'? If not, let's roll w/ the after

You mean the color of comment/like icons/labels?

If so, we have a ton of different greys :)
[https://gyazo.com/5aeda98cce45f83f3fc1670770797219](https://gyazo.com/5aeda98cce45f83f3fc1670770797219)

I was going for the same color as the rest of the icons (like in My Site tab). Would you like me to change the color to something else?

> I'm not sure about making the entire notification string the same dark shade. Can we maintain the separation of colors in strings between light/dark, but simply darken the parts that are too light? For example: https://cloudup.com/cp2yVHBpVhP

I looked at the Calypso for reference on this one. This is how the same notification looks there:
[https://gyazo.com/af96478c83c0c24795b8ccae44329aba](https://gyazo.com/af96478c83c0c24795b8ccae44329aba)

Shoul we go for a parity with Calypso or it doesn't really matter? (maybe @folletto has some thoughts on this one too).

**Comment by mattmiklic** [comment]
This is nice, the inconsistency on Android has been a pet peeve of mine for a while but there just hadn't been time yet to really audit how we're using color to make it more consistent. Big 👍 to this effort.

It would be helpful if there were an APK the designers could install to give this a test-run to check for edge cases that might need special attention, like the blue-on-blue toggle switch that @malinajirka noted above.

**Comment by etoledom** [comment]
> Also, I'd like to see if we can/should match similar on iOS.
@iamthomasbishop I we absolutely should!

We should think about Android and iOS apps as one consistent product because... aren't they?
Of course there are platform specific differences but things like colors and the experience in general should match completely.

I haven't seen any big inconsistency on iOS but there are some low contrast details for sure.

It might be good to finish this changes on Android, and then apply the final result on iOS.
What do you think?

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java [hunk 1]
```diff
@@ -229,6 +229,7 @@ public boolean onPreferenceChange(final Preference preference, final Object valu
                         int index = Integer.parseInt(value.toString());
                         CharSequence[] entries = editorTypePreference.getEntries();
                         editorTypePreference.setSummary(entries[index]);
+                        editorTypePreference.setValue(value.toString());
```
[the remark this answers is not in the document]
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java]: Yes, you are right, the system will save preference, but we are using custom `DetailListPreference` (for style reasons) which will not pick the change automatically so we have to manually set the value.

**Comment by iamthomasbishop** [comment]
@khaykov regarding your last comments above:

Placeholder text color (gray-darken-10) looks good 👍 

> SEND button is always active, on Android at least.

Fair enough, let's roll w/ it.

@khaykov Again, _great work_! I'm so happy that we're focused on consistency and accessibility.

> We are using blue_wordpress (#0087be) for similair controls, so I changed dropdown icon color to it.

👍 

> I was going for the same color as the rest of the icons (like in My Site tab). Would you like me to change the color to something else?

Sounds good - let's roll w/ it :)

> I looked at the Calypso for reference on this one. This is how the same notification looks there...Shoul we go for a parity with Calypso or it doesn't really matter?

Yea, that's fine - let's match Calypso there.

-----

Also good catches from @malinajirka . I'm not exactly sure what we should do about the blue-on-blue toggle...I'm not seeing the same thing if I go to `Me » Notifications`: 

screenshot_20180419-151114

-----

> Of course there are platform specific differences but things like colors and the experience in general should match completely.

> It might be good to finish this changes on Android, and then apply the final result on iOS.
What do you think?

@etoledom - I'm not super concerned with _every single color_ of every component matching exactly across platforms (yet! A new design language will give us the opportunity to get all of our ducks in a row across all platforms - so we will be addressing that in a wider way soon!). 

With that said, I do think within each app colors of components should be consistent (and contrast-tested, of course) across all screens within the app.

**Comment by khaykov** [comment]
@iamthomasbishop thanks! If you notice any weird colors after we merge this PR let me know and I'll fix them asap :)

@malinajirka I think I fixed all the issues you mentioned and ready for another round :)

**Comment by malinajirka** [comment]
Thanks @khaykov for fixing all the issues;)! 

I have found few more issues. I'm sorry I missed them during the first round:(.

1. Sharing buttons preferences are not consistent with other preferences screens.
sharing_buttons


2. The plugins icon doesn't use the correct shade of blue
mysite


3., 4., 5. These might still have contrast issues
User Detail
user_detail

Sharing
sharing

Plugin Detail
plugin_detail

Thanks!!!

**Comment by khaykov** [comment]
@malinajirka thank you so much for taking a look! I updated contrast at Sharing preferences screen, but I'll leave the overall design for another issue (#7682)  - it's custom Preference screen, and redesigning it is probably will require some consideration.

The User Detail screen is actually supposed to be like this - we are reducing the opacity of the container to indicate that it's disabled.

**Comment by malinajirka** [comment]
Thank you so much for this PR. Looks good!
